blog/2021-05-19-monero-transaction.md

```diff
@@ -0,0 +1,280 @@
+CLSAG _seeds_ the algorithm by first generating `c_π+1` based on `⍺`.
+More specifically, `L_π+1 = ⍺ * G`, which is successfully closed by `r_π`:
+first rearranging to get `⍺ = r_π + c_π * sk_π`; and then substituting in `L_π+1` giving `L_π+1 = r_π * G + c_π * sk_π * G = r_π * G + c_π * pk_π * G`, exactly what we'd expect from the definition.
+
+In Monero's CLSAG we instead calculate `r_π = ⍺ - c_π * mu_P * sk_π - c_π * mu_C * z`, where `z` is the difference between the real commitment's blinding factor and the pseudo output's commitment blinding factor.
+If we rearrange as above we get `⍺ = r_π + c_π * mu_P * sk_π + c_π * mu_C * z`, which when substituted in `L_π+1` gives `L_π+1 = r_π * G + c_π * mu_P * sk_π * G + c_π * mu_C * z * G` which is equal to `L_π+1 = r_π * G + c_π * mu_P * pk_π + c_π * mu_C * (C_π - C_pseudo)` if and only if `z * G = C_π - C_pseudo`.
+Since the real input's commitment and the pseudo output's commitment hide the same amount `C_π - C_pseudo = z_0 * G + a * H - (z_1 * G + a * H) = (z_0 - z_1) * G`.
+Therefore we can simplify our condition to `z * G = (z_0 - z_1) * G`, which is true given our definition of `z` above.
+
+The differences in the `R` term can be described similarly, but we direct the interested reader to our [implementation of Monero](https://github.com/comit-network/monero-rs/blob/b41ae9c32cb435a0f0c8e03f450dd248507c43b5/src/clsag.rs)'s CLSAG for more details.
+
+###### Constants `mu_P` and `mu_C`
+
+These constants and their role in CLSAG were introduced in the previous section.
+They are computed as
+
+```
+mu_P = H_n("CLSAG_agg_0" || ring || commitment_ring || I || D_inv_8 || pseudo_output_commitment);
+mu_C = H_n("CLSAG_agg_1" || ring || commitment_ring || I || D_inv_8 || pseudo_output_commitment);
+```
+
+where `I` is the key image `I = sk_π * H_p(pk_π)`; and `D_inv_8 = z * H_p(pk_π) * INV_EIGHT`.
+
+###### Domain tag and public parameters
+
+The actual components that remain constant between iterations of CLSAG are
+
+```
+c_i+1 = H_n("CLSAG_round" || ring || commitment_ring || pseudo_output_commitment || transaction_hash || ...);
+```
+
+where the ellipsis represents the missing components `L` and `R` discussed above, which vary between iterations.
+
+<!-- TODO: Thomas to describe how the message hash is computed in separate subsection. Also, there is one place where I've said transaction_hash when referring to the message hash that is signed. I'm not sure if that's wrong/confusing
+-->
+
+## Outlook
+
+We have put this together hoping that this information is useful to others, particularly to those attempting to write their own Monero wallets.
+We ourselves are in the process of completing a minimal Monero wallet, which you can find [here](https://github.com/comit-network/xmr-btc-swap/tree/master/monero-wallet).
```

Comment:
If we want to link to something, we should link to the latest commit that implements the transaction signing code. The code in `master` is pretty boring :sweat_smile:

Reply:
Good point!